lua/apisix/plugins/ip-restriction.lua

```diff
@@ -0,0 +1,145 @@
+-- Ref: https://github.com/Kong/kong/blob/master/kong/plugins/ip-restriction
+
+local ipairs = ipairs
+local type = type
+
+local core = require("apisix.core")
+local iputils = require("resty.iputils")
+
+local schema = {
+    type = "object",
+    properties = {
+        whitelist = {
+            type = "array",
+            items = {type = "string"},
+            minItems = 1
+        },
+        blacklist = {
+            type = "array",
+            items = {type = "string"},
+            minItems = 1
+        }
+    },
+    oneOf = {
```

Comment:
if we need whitelist and blacklist at the same time, what should we do?

Reply:
The main objective of this plugin just implements IP `blacklist` and `whitelist`, this means it just classify the IP address space to 2 set, `ALLOW` and `DENY`.

If we need whitelist and blacklist at the same time, we need  add default policy, maybe priority even need added, it's too complex, this may a `ACL`  to implements.

Reply:
got it. the whitelist and blacklist are mutually exclusive in this plugin.